File: src/test/java/br/com/jiratorio/integration/userconfig/FindUserConfigIntegrationTest.java
package br.com.jiratorio.integration.userconfig;

import br.com.jiratorio.base.Authenticator;
import br.com.jiratorio.domain.ChartType;
import br.com.jiratorio.domain.entity.UserConfig;
import br.com.jiratorio.factory.entity.UserConfigFactory;
import io.restassured.RestAssured;
import org.apache.http.HttpStatus;
import org.hamcrest.Matchers;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

@Tag("integration")
@ExtendWith(SpringExtension.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class FindUserConfigIntegrationTest {

    private final Authenticator authenticator;

    private final UserConfigFactory userConfigFactory;

    @Autowired
    FindUserConfigIntegrationTest(final Authenticator authenticator,
                                  final UserConfigFactory userConfigFactory) {
        this.authenticator = authenticator;
        this.userConfigFactory = userConfigFactory;
    }

    @Test
    void findCurrentUserConfig() {
        UserConfig userConfig = authenticator.withDefaultUser(userConfigFactory::create);

        // @formatter:off
        RestAssured
                .given()
                    .log().all()
                    .header(authenticator.defaultUserHeader())
                .when()
                    .get("/users/me/configs")
                .then()
                    .log().all()
                    .statusCode(HttpStatus.SC_OK)
                    .body("state", Matchers.is(userConfig.getState()))
                    .body("city", Matchers.is(userConfig.getCity()))
                    .body("holidayToken", Matchers.is(userConfig.getHolidayToken()))
                    .body("leadTimeChartType", Matchers.is(userConfig.getLeadTimeChartType().name()))
                    .body("throughputChartType", Matchers.is(userConfig.getThroughputChartType().name()));
        // @formatter:on
    }

    @Test
    void findNotCustomizedCurrentUserConfig() {
        // @formatter:off
        RestAssured
                .given()
                    .log().all()
                    .header(authenticator.defaultUserHeader())
                .when()
                    .get("/users/me/configs")
                .then()
                    .log().all()
                    .statusCode(HttpStatus.SC_OK)
                    .body("state", Matchers.nullValue())
                    .body("city", Matchers.nullValue())
                    .body("holidayToken", Matchers.nullValue())
                    .body("leadTimeChartType", Matchers.is(ChartType.BAR.name()))
                    .body("throughputChartType", Matchers.is(ChartType.DOUGHNUT.name()));
        // @formatter:on
    }
}

File: src/test/kotlin/br/com/jiratorio/integration/userconfig/FindUserConfigIntegrationTest.kt
package br.com.jiratorio.integration.userconfig

import br.com.jiratorio.base.Authenticator
import br.com.jiratorio.domain.chart.ChartType
import br.com.jiratorio.dsl.restAssured
import br.com.jiratorio.factory.entity.UserConfigFactory
import org.apache.http.HttpStatus
import org.hamcrest.Matchers.equalTo
import org.hamcrest.Matchers.nullValue
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension

@Tag("integration")
@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
internal class FindUserConfigIntegrationTest @Autowired constructor(
    private val authenticator: Authenticator,
    private val userConfigFactory: UserConfigFactory
) {

    @Test
    fun `find current user config`() {
        val userConfig = authenticator.withDefaultUser { userConfigFactory.create() }

        restAssured {
            given {
                header(authenticator.defaultUserHeader())
            }
            on {
                get("/users/me/configs")
            }
            then {
                statusCode(HttpStatus.SC_OK)
                body("state", equalTo(userConfig.state))
                body("city", equalTo(userConfig.city))
                body("holidayToken", equalTo(userConfig.holidayToken))
                body("leadTimeChartType", equalTo(userConfig.leadTimeChartType.name))
                body("throughputChartType", equalTo(userConfig.throughputChartType.name))
            }
        }
    }

    @Test
    fun `find not customized current user config`() {
        restAssured {
            given {
                header(authenticator.defaultUserHeader())
            }
            on {
                get("/users/me/configs")
            }
            then {
                statusCode(HttpStatus.SC_OK)
                body("state", nullValue())
                body("city", nullValue())
                body("holidayToken", nullValue())
                body("leadTimeChartType", equalTo(ChartType.BAR.name))
                body("throughputChartType", equalTo(ChartType.DOUGHNUT.name))
            }
        }
    }
}
